crates/dids/src/method/jwk.rs

```diff
@@ -0,0 +1,120 @@
+use crate::did::Did;
+use crate::method::{DidMethod, DidMethodError};
+use crate::resolver::DidResolutionResult;
+use async_trait::async_trait;
+use crypto::key::{Key, KeyType};
+use crypto::key_manager::KeyManager;
+use did_jwk::DIDJWK as SpruceDidJwkMethod;
+use ssi_dids::did_resolve::{DIDResolver, ResolutionInputMetadata};
+use ssi_dids::{DIDMethod, Source};
+use std::sync::Arc;
+
+/// Concrete implementation for a did:jwk DID
+pub struct DidJwk {
+    uri: String,
+    key_manager: Arc<dyn KeyManager>,
+}
+
+impl Did for DidJwk {
+    fn uri(&self) -> &str {
+        &self.uri
+    }
+
+    fn key_manager(&self) -> &Arc<dyn KeyManager> {
+        &self.key_manager
+    }
+}
+
+/// Options that can be used to create a did:jwk DID
+pub struct DidJwkCreateOptions {
+    pub key_type: KeyType,
+}
+
+#[async_trait]
+impl DidMethod<DidJwk, DidJwkCreateOptions> for DidJwk {
+    const NAME: &'static str = "jwk";
+
+    fn create(
+        key_manager: Arc<dyn KeyManager>,
+        options: DidJwkCreateOptions,
+    ) -> Result<DidJwk, DidMethodError> {
+        let key_alias = key_manager.generate_private_key(options.key_type)?;
+        let public_key =
+            key_manager
+                .get_public_key(&key_alias)?
+                .ok_or(DidMethodError::DidCreationFailure(
+                    "PublicKey not found".to_string(),
+                ))?;
+
+        let uri = SpruceDidJwkMethod
+            .generate(&Source::Key(&public_key.jwk()))
+            .ok_or(DidMethodError::DidCreationFailure(
+                "Failed to generate did:jwk".to_string(),
+            ))?;
+
+        Ok(DidJwk { uri, key_manager })
+    }
+
+    async fn resolve_uri(did_uri: &str) -> DidResolutionResult {
```

Comment:
There's a lot of code within TBD that use the term "did." Depending on where you're looking, it could mean one of two things:
1. The unique DID string
2. A structured representation of a DID

In this PR, I chose to always refer to the unique DID string as `did_uri` to hopefully help clarity while reading the code. 

Looking for feedback on this! 
Do you love it? Do you hate it? Happy to change if there's any strong feelings here.

Reply:
Makes sense, it's where we landed on the KT side (in most places).